origin-dapp-creator-server/src/middleware.js

```diff
@@ -11,8 +11,19 @@ export async function validateSubdomain(req, res, next) {
   const { address, config } = req.body
 
   if (config.subdomain) {
+    // Check subdomain doesn't contain invalid characters
+
+    // eslint-disable-next-line no-useless-escape
+    const subdomainRe = /[^a-zA-Z0-9\-]/
+    if (subdomainRe.test(config.subdomain)) {
+      return res.status(400).send('Subdomain contains invalid characters')
+    }
+
     // Check for subdomain blacklisting
     if (subdomainBlacklist.includes(config.subdomain.toLowerCase())) {
+      logger.warn(
```

Comment:
nit: should this be an error rather than a warning ? same comment for similar logging below.

Reply:
I normally do warnings for things that are the users fault like this, but one below needs to be updated, good spot!